web-ui/src/main/resources/catalog/components/common/map/externalviewer.js

```diff
@@ -0,0 +1,133 @@
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+
+(function() {
+  goog.provide('gn_external_viewer');
+
+  goog.require('gn_ows');
+
+
+  var module = angular.module('gn_external_viewer', [
+    'gn_ows'
+  ]);
+
+  /**
+   * @ngdoc service
+   * @kind function
+   * @name gn_external_viewer.service:gnExternalViewer
+   *
+   * @description
+   * The `gnExternalViewer` service is responsible for redirecting users to
+   * a third party viewer app when this feature is enabled in the UI settings.
+   */
+  module.service('gnExternalViewer', [
+    '$window',
+    'gnGlobalSettings',
+    '$location',
+    function($window, gnGlobalSettings, $location) {
+      /**
+       * Url pattern for metadata page
+       * @type {string}
+       */
+      var baseMdUrl = $location.absUrl().split('#')[0] + '#/metadata/';
+
+      /**
+       * Settings related to external viewer
+       * @type {Object}
+       */
+      var settings = gnGlobalSettings.gnCfg.mods.map.externalViewer;
+
+      return {
+        /**
+         * @ngdoc method
+         * @methodOf gn_external_viewer.service:gnExternalViewer
+         * @name gnExternalViewer#isEnabled
+         *
+         * @description
+         * Simple check against UI settings to see if an external viewer is
+         * enabled.
+         *
+         * @return {boolean} true if enabled
+         */
+        isEnabled: function() {
+          return !!settings.enabled;
+        },
+
+        /**
+         * @ngdoc method
+         * @methodOf gn_external_viewer.service:gnExternalViewer
+         * @name gnExternalViewer#openService
+         *
+         * @description
+         * Does the actual redirection inc. url parameters replacement.
+         *
+         * @param {Object} md expected properties: uuid, id, url
+         * @param {Object} service expected properties: url, name, type, title
+         */
+        viewService: function(md, service) {
+          if (!this.isEnabled()) { return; }
+
+          md.url = md.uuid ? baseMdUrl + md.uuid : '';
+
+          this._toView.push({
+            md: md,
+            service: service
+          });
+
+          // ask for a commit (will be executed once all services are requested
+          var commitFunction = this._commit.bind(this);
+          setTimeout(commitFunction);
+        },
+
+        _toView: [],
+
+        /**
+         * internal method: called once all services have been requested
+         */
+        _commit: function() {
+          if (!this._toView.length) { return; }
+
+          var getValues = function(object, key) {
+            return this._toView.map(function (entry) {
+              return entry[object][key] || '';
+            }).join(settings.valuesSeparator || ',') || ''
+          }.bind(this);
+
+          var url = settings.urlTemplate
```

Comment:
Maybe we should add some kind of warning/notification somewhere (in the admin or here) when the external viewer is turned on and no URL is defined ?

Reply:
I guess I'll add "&& baseUrl" in the `isEnabled` function of the external viewer service, along with a precision in the admin settings text. Will make another PR for this ASAP.